app/graphql/mutations/create_grading.rb

```diff
@@ -23,153 +23,150 @@ class CreateGrading < ApplicationQuery
                }
              }
 
-    description 'Create grading for submission'
+    description "Create grading for submission"
 
     field :success, Boolean, null: false
 
     def resolve(_params)
       grade
       notify(
         :success,
-        I18n.t('mutations.create_grading.grade_recorded'),
-        I18n.t('mutations.create_grading.success_notification')
+        I18n.t("mutations.create_grading.grade_recorded"),
+        I18n.t("mutations.create_grading.success_notification")
       )
       { success: true }
     end
 
     class ValidateReviewData < GraphQL::Schema::Validator
       include ValidatorCombinable
+
       def validate(_object, _context, value)
-        @submission_id = value[:submission_id]
+        @error = nil
         @submission = TimelineEvent.find_by(id: value[:submission_id])
         @checklist = value[:checklist]
-        @evaluation_criteria = @submission.evaluation_criteria
+        @evaluation_criteria = @submission&.evaluation_criteria
         @grades = value[:grades]
-        combine(
-          submission_exists,
-          submission_not_graded,
-          valid_evaluation_criteria,
-          right_shape_for_checklist,
-          checklist_data_is_not_mutated,
-          valid_grading
-        )
-      end
+        grade_hash = compute_grade_hash
```

Comment:
Why not make it `@grade_hash`. I see a lot of places where you are passing it around; 🤔

Reply:
Same problem, persistence of objects across multiple requests.

Reply:
Validators should avoid setting instance variables as much as possible. Validator instances are shared across requests.